analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -263,11 +263,38 @@ operation.Kind switch
         private static ExceptionState ExceptionCandidate(IOperationWrapperSonar operation) =>
             operation.Instance.Kind switch
             {
+                OperationKindEx.ArrayElementReference => ExceptionState.UnknownException, // ToDo: be explicit regarding exception type
+                OperationKindEx.Conversion => ConversionExceptionCandidate(operation),
+                OperationKindEx.DynamicIndexerAccess => ExceptionState.UnknownException, // ToDo: be explicit regarding exception type
+                OperationKindEx.DynamicInvocation => ExceptionState.UnknownException, // ToDo: be explicit regarding exception type
+                OperationKindEx.DynamicMemberReference => ExceptionState.UnknownException, // ToDo: be explicit regarding exception type
+                OperationKindEx.DynamicObjectCreation => ExceptionState.UnknownException,
+                OperationKindEx.EventReference => ExceptionCandidate(IMemberReferenceOperationWrapper.FromOperation(operation.Instance)),
+                OperationKindEx.FieldReference => ExceptionCandidate(IMemberReferenceOperationWrapper.FromOperation(operation.Instance)),
                 OperationKindEx.Invocation => ExceptionState.UnknownException,
-                // ToDo: Support other operations like field/property access on non-static non-this, conversions and so on. See docs for list of operations.
+                OperationKindEx.MethodReference => ExceptionCandidate(IMemberReferenceOperationWrapper.FromOperation(operation.Instance)),
+                OperationKindEx.ObjectCreation => operation.Instance.Type.DerivesFrom(KnownType.System_Exception) ? null : ExceptionState.UnknownException, // Exception constructors shouldn't throw
+                OperationKindEx.PropertyReference => ExceptionCandidate(IMemberReferenceOperationWrapper.FromOperation(operation.Instance)),
                 _ => null
             };
 
+        private static ExceptionState ConversionExceptionCandidate(IOperationWrapperSonar operation)
```

Comment:
Let's change the signature a little to to the conversion directly on invocation side. It will simplify the implementation to a single ternary, similar to the similar method below.

I know it will allocate one more struct on the stack for implicit conversion. I consider that cheap enough.
```suggestion
        private static ExceptionState ConversionExceptionCandidate(IConversionOperationWrapper conversion)
```

Reply:
The problem with this is that the `IConversionOperationWrapper` does not have `IsImplicit` property and that will split the logic between this method and the caller. I think it's better to keep it like this.